### DISCUSSION TITLE
Multi relation pagination

### DISCUSSION THREAD

**Discussion by tigawanna** [pocketbase/pocketbase#2783] [Q&A] [answered]
i see the muli relation field returns an array , 
is there a way to paginate the results in that specific array ?

For context am trying to figure out the best way to represent user and followers/following data.
this approach involves having each user have  a following/follower field with multiple relations .

The other approach i've tried was having a follwers collection that will store user and follower field , but i feel like there must be a more effecient way 
any advice

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not sure I follow. Could you provide an example?

> i see the muli relation field returns an array , is there a way to paginate the results in that specific array ?

Do you mean in the relation expand? If yes, then No, there is no pagination for the records within the `expand` field.

If you want pagination, you can send a separate request to the specific collection (and maybe a `filter` specifying the ids you want to fetch).

**Comment by tigawanna** [comment]
the schema would look something like this
```ts
export interface FollowingRecord {
    collectionId: string
    collectionName: string
    created: string
    expand: Expand
    following:( user relation:multi )
    id: string
    updated: string
    user:( user relation:single )
}

```

the query would also look like this
```ts
        const record = await pb.collection('following')
             .getFirstListItem<FollowingRecord>(`user.id = "${user.id}"`, {
            expand: 'following',
        });
```
And the reposnse would look something like this 
```ts
export interface FollowingRecord {
    collectionId: string
    collectionName: string
    created: string
    expand: Expand
    following: string[]
    id: string
    updated: string
    user: string
}

export interface Expand {
    following: PBUserRecord[]
}
```
I actually had 2 quetsion on how multi relations work :
1.  How do i update it , if - were to follow another user and do 

```ts
        return pb.collection('followers').update(follow_id, {
             following:new_user_id
        });
```
Will this merge this with the existing ones or do i have to have the whole array and pass it back in with the new_user_id added to the old array

2. whether there's a way of paginating over the following field since it has an array of user ids which you answered no to

**Comment by ganigeorgiev** [comment]
> Will this merge this with the existing ones or do i have to have the whole array and pass it back in with the new_user_id added to the old array

You have to pass the entire array, or use the `+` (append), `-` (remove) modifiers:

```js
await pb.collection('followers').update(follow_id, {
    // add a single user id into the "following" relations array
    // (pass an array if you want to append more than one id)
    "following+": new_user_id
});
```

> whether there's a way of paginating over the following field since it has an array of user ids which you answered no to

I still don't understand what exactly do you mean by _"paginating over the following field"_. What is the expected pagination output for example if there are more than one record with `following` relations? The `relation` field value is returned as it is in the record response (it is either a single id or array of ids).

If you want pagination for the `following` collection records, you need to send a separate `getList()` request.

**Reply by tigawanna** [reply]
> (it is either a single id or array of ids).

this is not a feature request but out of couriosity , wanted to know if the array of ids could be returned in chunks of let's say 20 , and on every subsequent request user would pass  the last id of the previous chunk as the cursor for where  the next chunk should start from , the only field that would change would be the 'following field because we're getting one record which has multiple following fields as an array

**Reply by ganigeorgiev** [reply]
It is not possible and I'm not sure if it can be implemented as it would be ambiguous. How do you specify the range of the field in the first place?

I guess one possible request format could be to reuse the `fields` parameter and extends its syntax to allow specifying "range", like:

```js
await pb.collection('followers').update(follow_id, {
    "fields": "id,name,following:0-20"
});
```

Something similar was requested before for large `text` fields and could be extended for array values, but for now this exists just in my todo to consider it at some later stage.
